hypothesis-python/tests/common/debug.py

```diff
@@ -51,16 +56,14 @@ def wrapped_condition(x):
     )
     def inner(x):
         if wrapped_condition(x):
-            result[:] = [x]
+            nonlocal result
+            result = x
             raise Found
 
-    definition.validate()
-    runtime = []
-    result = []
```

Comment:
I can only assume that these were declared as single-element lists so that they could be referenced from function-scope. I found this very confusing (to the point where I'm still unsure there's not some deeper reason here), so I've rewritten with `nonlocal` as a drive-by change.